mod/discussions/views/default/resources/discussion/owner.php

```diff
@@ -3,16 +3,22 @@
 $guid = elgg_extract('owner_guid', $vars);
 elgg_set_page_owner_guid($guid);
 
+elgg_entity_gatekeeper($guid);
 elgg_group_gatekeeper();
 
-$group = get_entity($guid);
-if (!elgg_instanceof($group, 'group')) {
+$container = get_entity($guid);
+$type = $container->type;
+$subtype = $container->getSubtype();
+if (!discussion_is_context_allowed($container->type, $container->getSubtype())) {
```

Comment:
@juho-jaakkola I don't how we could do this (404 the page if the container isn't an allowed type) without a new hook or a config array of some kind. We can't just check permissions.